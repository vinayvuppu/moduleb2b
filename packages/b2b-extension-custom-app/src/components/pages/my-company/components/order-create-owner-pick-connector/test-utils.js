/* eslint-disable import/prefer-default-export */
import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import flatMap from 'lodash.flatmap';
import intersection from 'lodash.intersection';
import FetchCustomersQuery from './order-create-customer-pick-connector.graphql';

export const createCustomersFetchQueryMock = (stores = []) => {
  const whereClause =
    stores.length === 0
      ? undefined
      : `(stores(key = "${stores.join(',')}") or stores is empty)`;
  const customers = [
    {
      id: 'customer-1',
      version: 4,
      firstName: 'Anna',
      lastName: 'Banana',
      email: 'annabanana@blablub.com',
      customerNumber: '111',
      companyName: null,
      customerGroup: null,
      stores: [
        {
          id: 'store-2',
          key: 'usa',
          nameAllLocales: [{ locale: 'en-GB', value: 'USA' }],
        },
      ],
      addresses: [
        {
          id: '_ZqTe0es',
          firstName: '3333',
          streetName: 'Fakestreet',
          streetNumber: '23',
          country: 'DE',
          additionalAddressInfo: 'Additional address info2',
          additionalStreetInfo: 'Additional street info',
          email: null,
          phone: null,
          city: null,
          company: null,
          lastName: null,
          postalCode: null,
          region: null,
        },
      ],
    },
    {
      id: 'customer-2',
      version: 1,
      firstName: 'Ali',
      lastName: 'Baba',
      customerNumber: '235',
      email: 'alibaba@testest.de',
      companyName: null,
      customerGroup: null,
      stores: [
        {
          id: 'store-1',
          key: 'germany',
          nameAllLocales: [{ locale: 'en-GB', value: 'Germany' }],
        },
      ],
      addresses: [],
    },
    {
      id: 'customer-3',
      version: 3,
      customerNumber: '128',
      firstName: 'Amin',
      lastName: 'Rocks',
      email: 'email@ct.de',
      companyName: null,
      customerGroup: null,
      stores: [],
      addresses: [
        {
          id: 'yUQGmqRG',
          firstName: 'first',
          lastName: 'last2',
          company: 'company',
          streetName: 'Street name',
          streetNumber: 'House number',
          city: 'city',
          postalCode: '66215',
          region: 'region',
          country: 'US',
          additionalAddressInfo: 'Additional address info2',
          additionalStreetInfo: 'Additional street info',
          email: 'email2@email.com',
          phone: 'phone',
        },
      ],
    },
  ].filter(customer => {
    // if we don't pass stores then we don't filter
    if (!stores) return true;

    const customerStoresArray = flatMap(customer.stores, s => s.key);
    return (
      // true if the customer belongs to one of the stores passed
      intersection(customerStoresArray, stores).length !== 0 ||
      // or the customer belongs to no store
      customerStoresArray.length === 0
    );
  });

  return {
    request: {
      variables: {
        target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
        projectKey: 'context-playground',
        limit: 20,
        locale: 'en',
        offset: 0,
        sort: ['createdAt desc'],
        where: whereClause,
      },
      query: FetchCustomersQuery,
    },
    result: {
      data: {
        customers: {
          count: 3,
          offset: 0,
          total: 3,
          results: customers,
        },
      },
    },
  };
};
