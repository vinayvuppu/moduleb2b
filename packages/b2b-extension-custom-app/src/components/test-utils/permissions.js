import { PERMISSIONS } from '../../constants';

// eslint-disable-next-line import/prefer-default-export
export const allTestPermissions = Object.keys(PERMISSIONS).reduce(
  (permissionsMap, permissionKey) => ({
    ...permissionsMap,
    [`can${permissionKey}`]: true,
  }),
  {}
);
