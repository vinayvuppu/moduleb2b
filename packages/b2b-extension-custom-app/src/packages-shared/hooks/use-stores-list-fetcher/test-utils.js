import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import { FetchStoresQuery } from './stores-list-query.graphql';

// eslint-disable-next-line import/prefer-default-export
export const createFetchStoresListQueryMock = (custom = {}) => ({
  request: {
    variables: {
      target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
      limit: 20,
      offset: 0,
      sort: `createdAt desc`,
    },
    ...custom.request,
    query: FetchStoresQuery,
  },
  result: {
    data: {
      stores: {
        count: 2,
        offset: 0,
        total: 2,
        results: [
          {
            id: 'store-1',
            key: 'germany',
            version: 1,
            nameAllLocales: [{ locale: 'en-GB', value: 'Germany' }],
            createdAt: '2018-12-19T15:26:51.174Z',
            lastModifiedAt: '2019-03-01T09:00:51.174Z',
          },
          {
            id: 'store-2',
            key: 'usa',
            version: 1,
            nameAllLocales: [{ locale: 'en-GB', value: 'USA' }],
            createdAt: '2019-06-10T15:00:51.140Z',
            lastModifiedAt: '2019-08-30T10:26:51.140Z',
          },
        ],
      },
    },
    ...custom.result,
  },
});
