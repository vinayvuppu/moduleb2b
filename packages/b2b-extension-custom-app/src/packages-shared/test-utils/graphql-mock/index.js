export { createRosie, createGraphqlMock } from './graphql-mock';
