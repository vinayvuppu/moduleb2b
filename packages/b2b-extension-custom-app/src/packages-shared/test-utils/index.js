import crypto from 'crypto';
import { withProps, compose, branch, renderComponent } from 'recompose';

export { createGraphqlMock, createRosie } from './graphql-mock';

export const intlMock = {
  formatMessage: jest.fn(message => message.id),
  formatDate: jest.fn(date => {
    const dateObj = new Date(date);
    const day = dateObj.getUTCDate();
    const month = dateObj.getUTCMonth() + 1;
    const year = dateObj.getUTCFullYear();
    return `${day}.${month}.${year}`;
  }),
  formatTime: jest.fn(date => {
    const dateObj = new Date(date);
    const hours = dateObj.getUTCHours();
    const minutes = dateObj.getUTCMinutes();
    return `${hours}:${minutes}`;
  }),
  formatRelative: jest.fn(() => 'xxx'),
  formatNumber: jest.fn((number, currency) =>
    currency?.currency
      ? `${currency.currency} ${number.toString()}`
      : number.toString()
  ),
  formatPlural: jest.fn(() => 'xxx'),
  formatHTMLMessage: jest.fn(() => 'xxx'),
  now: jest.fn(() => 'xxx'),
  locale: 'en',
};

export const routerMock = {
  go: jest.fn(() => {}),
  goBack: jest.fn(() => {}),
  goForward: jest.fn(() => {}),
  push: jest.fn(() => {}),
  replace: jest.fn(() => {}),
  isActive: jest.fn(() => {}),
  createHref: jest.fn(() => {}),
  setRouteLeaveHook: jest.fn(() => {}),
};

export function createMountOptions(options = {}) {
  const { intl, router } = options;
  return {
    context: {
      intl: { ...intlMock, ...intl },
      router: { ...routerMock, ...router },
      store: {
        getState: () => ({
          application: {
            products: {
              currentVisible: {
                productType: {
                  obj: {
                    name: 'foo',
                  },
                },
              },
            },
          },
        }),
        subscribe() {},
        dispatch() {},
      },
    },
  };
}

export function promisedTimeout(cb, timeout = 0) {
  const p = new Promise(resolve => {
    setTimeout(() => {
      resolve(cb());
    }, timeout);
  });
  return p;
}

export const randomId = () => crypto.randomBytes(16).toString('hex');

export const mockPermissions = ({ isAuthorized = true } = {}) => {
  const injectAuthorized = (
    demandedPermissions,
    options,
    propName = 'isAuthorized'
  ) => withProps({ [propName]: isAuthorized });

  const branchOnPermissions = (
    demandedPermissions,
    FallbackComponent,
    options
  ) =>
    compose(
      injectAuthorized(demandedPermissions, options),
      branch(() => !isAuthorized, renderComponent(FallbackComponent))
    );
  return {
    injectAuthorized,
    branchOnPermissions,
    permissions: {},
  };
};

export const mockBranchOnFeatureToggle = ({
  isFeatureToggledOn = true,
} = {}) => (flags, FallbackComponent) =>
  branch(() => !isFeatureToggledOn, renderComponent(FallbackComponent));

export const mockComponent = displayName => {
  const MockedComponent = ({ children }) => children || null;
  MockedComponent.displayName = displayName;
  return MockedComponent;
};

export const mockConnector = (displayName, childrenProps = {}) => {
  const MockedConnector = jest.fn(({ children }) => children(childrenProps));
  MockedConnector.displayName = displayName;
  return MockedConnector;
};

export const mockContext = displayName => {
  let providedValue;

  const ProviderComponent = ({ value, children }) => {
    providedValue = value;
    return children;
  };
  ProviderComponent.displayName = `${displayName}.Provider`;
  const ConsumerComponent = ({ children }) => children(providedValue);
  ConsumerComponent.displayName = `${displayName}.Consumer`;

  return {
    Provider: ProviderComponent,
    Consumer: ConsumerComponent,
  };
};

export const mockReactIntl = () => ({
  ...require.requireActual('react-intl'),
  FormattedMessage: ({ id }) => `formatted-${id}`,
  injectIntl: withProps({ intl: intlMock }),
});

export const waitForPromises = () =>
  new Promise(resolve => setImmediate(resolve));
