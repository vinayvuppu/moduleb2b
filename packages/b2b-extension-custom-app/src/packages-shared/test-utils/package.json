{
  "name": "@commercetools-local/test-utils",
  "version": "1.0.0",
  "description": "",
  "license": "Proprietary",
  "private": true,
  "main": "./index.js",
  "dependencies": {
    "@commercetools-frontend/constants": "15.3.4",
    "@commercetools-frontend/notifications": "15.3.4",
    "@commercetools-local/utils": "*",
    "faker": "4.1.0",
    "graphql": "14.5.8",
    "graphql-tools": "4.0.4",
    "lodash.merge": "4.6.2",
    "prop-types": "15.7.2",
    "recompose": "0.26.0",
    "rosie": "2.0.1"
  }
}
