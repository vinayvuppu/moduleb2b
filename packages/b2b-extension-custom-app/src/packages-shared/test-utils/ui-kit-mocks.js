/* eslint-disable global-require, react/prop-types, react/display-name */
jest.mock('@commercetools-frontend/ui-kit', () => {
  const React = require('react');
  class MockedTable extends React.Component {
    componentDidMount() {
      if (this.props.registerMeasurementCache) {
        this.props.registerMeasurementCache({
          clearAll: () => {},
        });
      }
    }
    render() {
      return (
        <div>
          <table>
            <thead>
              <tr>
                {this.props.columns.map(column => (
                  <td key={column.key}>{column.label}</td>
                ))}
              </tr>
            </thead>
            <tbody>
              {this.props.items.map((item, rowIndex) => (
                <tr key={rowIndex}>
                  {this.props.columns.map(column => (
                    <td
                      key={column.key}
                      data-testid={`cell-${rowIndex}-${column.key}`}
                      onClick={() =>
                        this.props.onRowClick
                          ? this.props.onRowClick(column.key, rowIndex)
                          : null
                      }
                    >
                      {this.props.itemRenderer({
                        rowIndex,
                        columnKey: column.key,
                      })}
                    </td>
                  ))}
                </tr>
              ))}
            </tbody>
          </table>
          {this.props.children}
        </div>
      );
    }
  }
  return {
    ...jest.requireActual('@commercetools-frontend/ui-kit'),
    Table: MockedTable,
  };
});
