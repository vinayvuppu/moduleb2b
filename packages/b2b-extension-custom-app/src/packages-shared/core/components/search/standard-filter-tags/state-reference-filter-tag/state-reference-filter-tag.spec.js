import React from 'react';
import { shallow } from 'enzyme';
import { intlMock } from '../../../../../test-utils';
import createStateFilterTag from './state-reference-filter-tag';

describe('export', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<div>{createStateFilterTag('OrderState')}</div>, {
      context: { intl: intlMock },
    });
  });
  it('should export composed component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
