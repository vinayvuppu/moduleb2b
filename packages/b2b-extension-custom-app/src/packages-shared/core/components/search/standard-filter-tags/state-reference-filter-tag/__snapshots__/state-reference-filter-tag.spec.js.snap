// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export should export composed component 1`] = `
<div>
  <Component />
</div>
`;
